### PR TITLE
FIX: filter conditionally excluded fields from wizard submission

### PR DESCRIPTION
**Pull request by fzngagan**

### CODE REVIEW

File: spec/requests/custom_wizard/steps_controller_spec.rb [hunk 1]
```diff
@@ -249,4 +249,41 @@
...
+      fields: {
+        step_1_field_1: "Condition will not pass"
+      }
+    }
+    wizard_id = response.parsed_body['wizard']['id']
+    wizard = CustomWizard::Wizard.create(wizard_id, user)
+    submission = wizard.submissions.last
+    expect(submission.keys).to include("step_2_field_1")
+
+    put '/w/super-mega-fun-wizard/steps/step_2.json', params: {
```
Comment (angusmcleod) [inline spec/requests/custom_wizard/steps_controller_spec.rb:280]: @fzngagan We need to break this up into two seperate tests.

**Review by angusmcleod — CHANGES_REQUESTED** [review]
This looks good, but I do think we need at least two seperate tests there, rather than one big test for multiple cases.

**Comment by fzngagan** [comment]
@angusmcleod 
My thinking is, we are testing exclusion here. So we need to confirm both inclusion and then exclusion.

**Comment by fzngagan** [comment]
Conclusion from the discussion.
re-submitting step 1, should get rid of the submission from step 2. Investigate why that's not the case.

**Comment by fzngagan** [comment]
@angusmcleod 
The issue was, that filtering needs to happen when the wizard is built **post submission being saved**. I've also simplified the spec.

**Review by angusmcleod — APPROVED** [review]

**Pull request merged**